zerver/lib/bugdown/__init__.py

```diff
@@ -1489,6 +1489,74 @@ def __init__(self, parser: Any) -> None:
         super().__init__(parser)
         parser.markdown.tab_length = 4
 
+class AdmonitionProcessor(admonition.AdmonitionProcessor):
+    """ Process !!! admonition blocks to produce colored status messages
+
+        This was modified from the standard AdmonitionProcessor to:
+
+        1. Change the element used from <p> to <blockquote> so that the
+           rendering in clients without additional CSS classes is reasonable.
+        2. Prefix classes added to the admonition blockquote with `admonition-`
+           to avoid conflicts with existing CSS classes in the stylesheet, such
+           as .info
+        3. Provide support for a fenced unindented syntax as an alternative to
+           the indented syntax:
+
+           !!! success "Message title"
+               Some content
+               Some more content
+
+           and
+
+           !!! success "Message title"
+           Some content
+           Some more content
+           !!!
+    """
+    RE_FINISH = re.compile(r'(?:\n|$)!!!\s*(?:\n|$)')
+
+    def get_class_and_title(self, match: Match) -> Tuple[str, str]:
+        klass, title = match.group(1).lower(), match.group(2)
+        klass = self.RE_SPACES.sub(' ', klass)
+        prefixed_klasses = ["admonition-{}".format(k) for k in klass.split(" ")]
+        klass = " ".join(prefixed_klasses)
+        return klass, title
+
+    def test(self, parent: Element, block: str) -> bool:
+        return self.RE.search(block)
```

Comment:
Is there a reason to remove support for multiblock admonitions?

Reply:
I may have misunderstood what was happening in there; can you help me understand what use case that was supporting before? I was having trouble getting multiple blocks grouped into one admonition (and reading `run()`, I don't see how it'd work?), but if it can be done, fantastic.

Reply:
If we rewrite the original test function, it'd appear like:

```
if self.RE.search(block):
	# Block starts with admonition syntax
	return True
if block.startswith('    ') and 'admonition' in sibling_block.get_classes()
	# Previous block was an admonition and current block is indented.
	return True
return False
```

Essentially, we'd handle the restriction you mention in the fenced implementation about not having any blank lines.

Reply:
Got it. I'll roll that back and remove the manual fencing code; my ignorance of the existing systems is showing here. If there's stuff already existing to handle it more robustly, all the better.